### PR TITLE
fix: formatting in vue files

### LINKED ISSUE #21: Formatting in Vue files is broken
Followup of https://github.com/simonhaenisch/prettier-plugin-organize-imports/issues/16#issuecomment-842605347

Version 2.1.0

In addition to the removal of new lines also comments gets destroyed and break the code!

Reproducible:

- Clone https://github.com/Shinigami92/vizscript
- Install deps
- Add a comment in a .vue file somewhere
- Run format (via VSCode or CLI)

### PR DESCRIPTION
**Pull request by Shinigami92**
closes #21

### CODE REVIEW

**Comment by Shinigami92** [comment]
@simonhaenisch So the test is failing, but I explored the code and it seems that your code doesn't make anything wrong.
I assume something maybe wrong on Prettier's side?!

When I `console.log` the produces result of your code, the code is correct except that there are no spaces in front of the commas in the import statements :thinking: 

So e.g. 
```diff
-import { defineComponent, ref } from "vue"
+import { defineComponent,ref } from "vue"
```

Maybe that is correlated with the offset?

Cause in the test exactly one character is removed

```diff
- // a test comment
+// a test commen
```

File: index.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ const organizeImports = (text, options) => {
 	}
 
 	try {
-		const fileName = !options.filepath || options.filepath.endsWith('.vue') ? 'file.ts' : options.filepath;
+		const fileName = !options.filepath || options.parentParser === 'vue' ? 'file.ts' : options.filepath;
```
Comment (simonhaenisch) [inline index.js:20]: Ok nice I didn't know about `parentParser`. 👍

**Comment by simonhaenisch** [comment]
> Maybe that is correlated with the offset?

There might be a mistake with how the changes are applied; debugging this is a bit of a time-consuming task though. Not sure I'll be able to do it today.

**Comment by simonhaenisch** [comment]
I started looking into it end of last week but didn't figure it out before I ran out of time. I've been trying to understand what the difference is between the language server that vs code runs and the one that I run for the plugin.

I've noticed that VS Code doesn't show the "Organize Imports" action if you create a new document without saving and set the language mode to "Vue". You actually have to write the file to disk first, then the organize action works.

So what I've been doing in the plugin is that for `.vue` files, I've been telling the TS language server that it's actually a typescript file (`file.ts`) which made the `organizeImports` function run but it didn't actually work correctly. This is where a fix is needed. I'll hopefully find some more time for this in the next couple days.

**Comment by simonhaenisch** [comment]
Thanks for the PR, I've added the test case in 4e547220f12d0cc354542b8a9c5fe3e421bdee85. Forgot to add you as a co-author though.

**Pull request closed without merging**